docs/releases/5.1.md

```diff
@@ -266,3 +266,43 @@ The global util will be removed in a future release. It is recommended that the
 ```
 
 Note: The `data-w-tag-options-value` is a JSON object serialised into string. Django's HTML escaping will handle it automatically when you use the `AdminTagWidget`, but if you are manually writing the attributes, be sure to use quotation marks correctly.
+
+### Header searching now relies on data attributes
+
+Previously the header search relied on inline scripts and a `window.headerSearch` global to activate the behaviour. This has now changed to a data attributes approach and the window global usage will be removed in a future major release.
+
+If you are using the documented Wagtail [viewsets](viewsets_reference), Snippets or `ModelAdmin` approaches to building custom admin views, there should be no change required.
+
+If you have customisations that manually declare or override `window.headerSearch`, here's how to adopt the new approach.
+
+**Before**
+
+```html
+<script>
+    window.headerSearch = {
+        url: '{{ my_async_results_url }}',
+        termInput: '#id_q',
+        targetOutput: '#some-results',
+    };
+</script>
+<form role="search">
+    <input type="text" name="q" id="id_q" />
+</form>
+<div id="some-results"></div>
+```
+
+**After**
+
+```html
+<form
+    role="search"
+    data-controller="w-swap"
+    data-action="change->w-swap#searchLazy input->w-swap#searchLazy"
+    data-w-swap-location-param="true"
```

Comment:
```suggestion
```

Might not be needed - see other comments & https://github.com/wagtail/wagtail/pull/9961#discussion_r1254384963

Reply:
Have reverted this approach in #9961  and no longer need the action param